### PR TITLE
Clean up temp dir after app tests

### PR DESCRIPTION
**Pull request by tcsc**
There is an intermittent issue where test will fail due to a non-empty
test directory at the end of a test. Temp dirs created by the `testing`
package are _not required to be empty_ at cleanup time, so something has
gone awry.

After looking at the code, it seems that an some Auth server instances
created as fixtures for several tests are never explicitly closed and
cleaned up when the test finishes. This leads to a hypothesis that
something in the audit writer is writing to the temp dir _while it is in
the process of being cleaned up_.

This patch add automatic cleanup to the Auth server test fixture, which
executes before the temp directory cleanup. The intention is to stop
any writes to the temp dir at the end of the test, so that it can be
cleaned up without error.

Also marks some not-implemented tests _skipped_ rather than _passed_, to
increase visibility.

### CODE REVIEW

File: lib/srv/app/server_test.go [hunk 1]
```diff
@@ -397,21 +398,25 @@ func TestAuthorizeWithLocks(t *testing.T) {
 // TestGetConfigForClient verifies that only the CAs of the requested cluster are returned.
 func TestGetConfigForClient(t *testing.T) {
 	// TODO(r0mant): Implement this.
+	t.Skip("Not Implemented")
```
Comment (codingllama) [inline lib/srv/app/server_test.go:401]: I wonder if it's even worth keeping these around. Ideally, IMO, we should get automated coverage reports, so we can see how each package stands - this should make it clear which critical packages are lacking in the test department.

**Review by codingllama — APPROVED** [review]
Thanks for the fix, those failures are kind of a pain.

**Review by russjones — APPROVED** [review]

File: lib/srv/app/server_test.go (hunk 1 above)
Comment (tcsc) [reply lib/srv/app/server_test.go:401]: That's definitely on my list (it's a _long_ list...). But, right now, at the very least they should not be reporting a pass.

Comment (codingllama) [reply lib/srv/app/server_test.go:401]: That's fair, thanks.

**Comment by greedy52** [comment]
I may still see this fails, or is this a different failure @tcsc ?

https://console.cloud.google.com/cloud-build/builds/72e047fc-c18d-49d7-a3ff-41160f376a22;step=0?project=ci-account

```
testing.go:967: TempDir RemoveAll cleanup: unlinkat /tmp/TestHandleConnection556806069/001/log/upload/streaming/default: directory not empty
--- FAIL: TestHandleConnection (2.91s)
exit status 1
Makefile:462: recipe for target 'test-go' failed
```

**Pull request merged**